### PR TITLE
Add manually triggerable Action to build simd Docker image

### PR DESCRIPTION
**Pull request by chatton**
## Description

I ran into a permissions error during the e2e tests which seems to have been a result of the original images not being built on a github runner. (they had been built on my local machine). The errors only occurred when using images that were not built on a github host.

See [this failure](https://github.com/cosmos/ibc-go/runs/7807503289?check_suite_focus=true) as a reference. 

See [this passing test](https://github.com/cosmos/ibc-go/runs/7808214404?check_suite_focus=true) which uses an image built by this workflow.

The test failures no longer occurred when the same image is built using this action instead.

Note: this workflow can be run with 

```bash
gh workflow run "Build Simd Image" --ref "cian/build-image-from-tag" -f tag=v3.0.0
```

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
🚀

**Review by crodriguezvega — COMMENTED** [review]
Should we add to the documentation for e2e the GitHub cli or UI steps needed to run this?

**Comment by chatton** [comment]
@crodriguezvega great suggestion I'll make an update to the e2e README

**Review by colin-axner — APPROVED** [review]
Nice!

File: e2e/README.md [hunk 1]
```diff
@@ -308,6 +309,25 @@ All tests will be run on different hosts.
 * Gas fees are set to zero to simply calcuations when asserting account balances.
 * When upgrading from e.g. v4 -> v5, in ibc-go, we cannot upgrade the go.mod under `e2e` since v5 will not yet exist. We need to upgrade it in a follow up PR.
 
+
+### Github Workflows
+
+#### Building and bushing a simd image.
```
Comment (crodriguezvega) [inline e2e/README.md:315]:
```suggestion
#### Building and pushing a `simd` image.
```

File: e2e/README.md [hunk 2]
```diff
@@ -308,6 +309,25 @@ All tests will be run on different hosts.
 * Gas fees are set to zero to simply calcuations when asserting account balances.
 * When upgrading from e.g. v4 -> v5, in ibc-go, we cannot upgrade the go.mod under `e2e` since v5 will not yet exist. We need to upgrade it in a follow up PR.
 
+
+### Github Workflows
```
Comment (crodriguezvega) [inline e2e/README.md:313]: nit: Officially the name of the company is with `H`.

```suggestion
### GitHub Workflows
```

File: e2e/README.md [hunk 3]
```diff
@@ -308,6 +309,25 @@ All tests will be run on different hosts.
...
+
+#### Building and bushing a simd image.
+
+If we ever need to manually build and push an image, we can do so with the [Build Simd Image](../.github/workflows/build-simd-image-from-tag.yml) github workflow.
```
Comment (crodriguezvega) [inline e2e/README.md:317]:
```suggestion
If we ever need to manually build and push an image, we can do so with the [Build Simd Image](../.github/workflows/build-simd-image-from-tag.yml) GitHub workflow.
```

File: e2e/README.md [hunk 4]
```diff
@@ -308,6 +309,25 @@ All tests will be run on different hosts.
...
+
+Thie can be triggered manually from the UI by navigating to
+
+`Actions` -> `Build Simd Image` -> `Run Workflow`
+
+And providing the git tag.
+
+Alternatively, the [gh](https://cli.github.com/) cli tool can be used to trigger this workflow.
```
Comment (crodriguezvega) [inline e2e/README.md:325]:
```suggestion
Alternatively, the [gh](https://cli.github.com/) CLI tool can be used to trigger this workflow.
```

File: e2e/README.md [hunk 5]
```diff
@@ -308,6 +309,25 @@ All tests will be run on different hosts.
...
 * When upgrading from e.g. v4 -> v5, in ibc-go, we cannot upgrade the go.mod under `e2e` since v5 will not yet exist. We need to upgrade it in a follow up PR.
 
+
+### Github Workflows
+
+#### Building and bushing a simd image.
+
+If we ever need to manually build and push an image, we can do so with the [Build Simd Image](../.github/workflows/build-simd-image-from-tag.yml) github workflow.
+
+Thie can be triggered manually from the UI by navigating to
```
Comment (crodriguezvega) [inline e2e/README.md:319]:
```suggestion
This can be triggered manually from the UI by navigating to
```

**Review by crodriguezvega — APPROVED** [review]
Great, @chatton! Thanks for adding the section to the docs!

**Pull request merged**